collector/monitors/update_global_index_monitor.go

```diff
@@ -0,0 +1,236 @@
+	UpdateGlobalIndexFailedTxSinceLastCheck     Metric = "update_global_index_failed_tx_since_last_check"
+	UpdateGlobalIndexGasWanted                  Metric = "update_global_index_gas_wanted"
+	UpdateGlobalIndexGasUsed                    Metric = "update_global_index_gas_used"
+	UpdateGlobalIndexUUSDFee                    Metric = "update_global_index_uusd_fee"
+)
+
+const threshold int = 10
+
+type UpdateGlobalIndexMonitor struct {
+	metrics          map[Metric]float64
+	ContractAddress  string
+	apiClient        *client.TerraLiteForTerra
+	logger           *logrus.Logger
+	lastMaxCheckedID int
+}
+
+func NewUpdateGlobalIndexMonitor(cfg config.CollectorConfig) UpdateGlobalIndexMonitor {
+	m := UpdateGlobalIndexMonitor{
+		metrics:         make(map[Metric]float64),
+		ContractAddress: cfg.UpdateGlobalIndexBotAddress,
+		apiClient:       cfg.GetTerraClient(),
+		logger:          cfg.Logger,
+	}
+
+	return m
+}
+
+func (m UpdateGlobalIndexMonitor) Name() string {
+	return "UpdateGlobalIndexMonitor"
+}
+
+func (m *UpdateGlobalIndexMonitor) InitMetrics() {
+	m.metrics[UpdateGlobalIndexSuccessfulTxSinceLastCheck] = 0
+	m.metrics[UpdateGlobalIndexGasWanted] = 0
+	m.metrics[UpdateGlobalIndexGasUsed] = 0
+	m.metrics[UpdateGlobalIndexUUSDFee] = 0
+	m.metrics[UpdateGlobalIndexFailedTxSinceLastCheck] = 0
+}
+
+func (m *UpdateGlobalIndexMonitor) Handler(ctx context.Context) error {
+	var offset *float64
+	var fetchedTxs int
+	var firstCheck bool
+	if m.lastMaxCheckedID == 0 {
+		firstCheck = true
+	}
+
+	iterations := 0
+	var maxProcessedID int
+	var maxProcessedIDPerRequest int
+	var alreadyProcessedFound bool
+	m.InitMetrics()
+	for iterations < threshold {
+		p := transactions.GetV1TxsParams{}
+		p.SetAccount(&m.ContractAddress)
+		p.SetContext(ctx)
+		p.SetOffset(offset)
+
+		resp, err := m.apiClient.Transactions.GetV1Txs(&p)
+		if err != nil {
+			return fmt.Errorf("failed to fetch transaction history for UpdateGlobalIndexBotContract account: %w", err)
+		}
+
+		maxProcessedIDPerRequest, alreadyProcessedFound = m.processTransactions(resp.Payload.Txs, m.lastMaxCheckedID)
+		fetchedTxs += len(resp.Payload.Txs)
+		maxProcessedID = maxInt(maxProcessedID, maxProcessedIDPerRequest)
+		if alreadyProcessedFound || firstCheck {
+			break
+		}
+		offset = resp.Payload.Next
+		iterations++
+	}
+	m.lastMaxCheckedID = maxProcessedID
+	if threshold == iterations {
+		m.logger.Warning("update global index processing stopped due to requests threshold - ", threshold)
+	}
+	m.logger.Infoln("update global index txs fetched:", fetchedTxs)
+	m.logger.Infoln("update global index state:", m.metrics)
+	return nil
+}
+
+func maxInt(a, b int) int {
+	if a > b {
+		return a
+	}
+	return b
+}
+
+func (m *UpdateGlobalIndexMonitor) processTransactions(txs []*models.GetTxListResultTxs, previousMaxCheckedID int) (newMaxCheckedID int, alreadyProcessedFound bool) {
+	for i, tx := range txs {
```

Comment:
Please add a comment that the txs are sorted in reverse order (from newer to older)

Reply:
done